service-print/src/main/java/org/oskari/print/loader/CommandLoadImageWMTS.java

```diff
@@ -190,10 +198,51 @@ private TileMatrix getTileMatrix(TileMatrixSet tms) throws IllegalArgumentExcept
     }
 
     private TileMatrixSet getTileMatrixSet() throws IllegalArgumentException {
+        List<TileMatrixSet> possibleTileMatrixSets = new ArrayList<>();
+
         for (TileMatrixSet tms : capabilities.getTileMatrixSets()) {
-            if (srs.equals(ProjectionHelper.shortSyntaxEpsg(tms.getCrs()))) {
-                return tms;
+            String key = tms.getTileMatrixMap().keySet().iterator().next();
```

Comment:
This gets the "first" identifier of a TileMatrix within the TileMatrixSet. Usually the value should be "0" or something like that. This most probably isn't what we are looking for.
Instead check that the TileMatrixSet supports the CRS of the request `if (srs.equals(ProjectionHelper.shortSyntaxEpsg(tms.getCrs())))` and only if it does then add it to the possibleTileMatrixSets list.